src/python/pants/backend/jvm/subsystems/zinc.py

```diff
@@ -151,6 +154,23 @@ def compiler_interface(self):
     """
     return self._zinc_factory._compiler_interface(self._products)
 
+  def snapshot(self, scheduler):
+    if self._snapshot is None:
+      buildroot = get_buildroot()
+      self._snapshot = scheduler.capture_snapshots((
+        PathGlobsAndRoot(
+          PathGlobs(
+            tuple(
+              fast_relpath(a, buildroot)
+              for a in (self.zinc, self.compiler_bridge, self.compiler_interface)
+            )
+          ),
+          buildroot,
+        ),
+      ))[0]
+    return self._snapshot
+
+  # TODO: Relativise (and sort)
```

Comment:
By definition, it is not possible to relativise these: they need to align with the host executing the `zinc` process.

Fixing this will require moving discovery of the JDK home into zinc, most likely.